tests/bufferingCache.it.spec.js

```diff
@@ -259,4 +260,216 @@ describe('buffering cache', () => {
       .then(() => done())
       .catch((err) => done(err || 'fail'));
   });
+
```

Comment:
Can you move all these new tests to a new `app.spec.js`?

And also, we still need a single test case that demonstrates the failure we had before, and verifies that it's fixed.